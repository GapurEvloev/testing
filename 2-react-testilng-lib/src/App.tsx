import React from "react";
import Navbar from "./components/Navbar/Navbar";
import AppRouter from "./router/AppRouter";

// function App() {
//   const [data, setData] = useState<null | {}>(null);
//   const [toggle, setToggle] = useState(false);
//   const [value, setValue] = useState("");
//
//   const onClick = () => setToggle(prev => !prev)
//
//   useEffect(() => {
//     setTimeout(() => {
//       setData({})
//     }, 100);
//   }, []);
//
//   return (
//     <div className="App">
//       <h1 data-testid="value-title">{value}</h1>
//       {toggle && (
//         <div data-testid="toggle-elem">toggle</div>
//       )}
//       {data && (
//         <div style={{color: 'red'}}>data</div>
//       )}
//       <h1>Hello world!</h1>
//       <button data-testid="toggle-button" onClick={onClick}>Click me</button>
//       <input onChange={(e) => setValue(e.target.value)} type="text" placeholder="input value...." />
//     </div>
//   );
// }

// integration tests with react router dom v6
const App = () => {
  return(
    <div>
      <Navbar/>
      <AppRouter/>
    </div>
  )
}

export default App;
