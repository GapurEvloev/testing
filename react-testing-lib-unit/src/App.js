import React, { useState, useEffect } from 'react';
import './App.css';

function App() {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState("");

  const handleClick = () => {
    setToggle(prev => !prev);
  }

  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 100);
  }, []);

  return (
    <div className="App">
      <h1>Hello world</h1>
      <h2>{data && <div className={'red'}>hello world2</div>}</h2>
      {toggle && <div data-testid="toggle-elem">toggle</div>}
      <button onClick={handleClick} data-testid="toggle-btn">Click me</button>
      <input onChange={e => setValue(e.target.value)} type="text" placeholder="enter your name..." />
      <div data-testid="value-elem">{value}</div>
    </div>
  );
}

export default App;
