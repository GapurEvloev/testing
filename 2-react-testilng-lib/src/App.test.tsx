import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe("TEST APP RTL", () => {
  test('renders learn react link', () => {
    render(<App />);
    const helloWorldElement = screen.getByText(/hello world!/i);
    const btnElement = screen.getByRole('button');
    const inputElement = screen.getByPlaceholderText(/input value/i);
    expect(helloWorldElement).toBeInTheDocument();
    expect(btnElement).toBeInTheDocument();
    expect(inputElement).toMatchSnapshot();
  });
});
