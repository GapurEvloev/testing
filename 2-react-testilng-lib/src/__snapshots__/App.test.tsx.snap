// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP RTL renders learn react link 1`] = `
<input
  placeholder="input value...."
  type="text"
/>
`;
